src/BaseUriPlugin.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Http\Promise\Promise;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+
+class BaseUriPlugin implements Plugin
+{
+    /**
+     * @var UriInterface
+     */
+    private $baseUri;
+
+    /**
+     * @param UriInterface $baseUri
+     */
+    public function __construct(UriInterface $baseUri)
+    {
+        $this->baseUri = $baseUri;
+    }
+
+    /**
+     * Handle the request and return the response coming from the next callable.
+     *
+     * @param RequestInterface $request
+     * @param callable         $next    Next middleware in the chain, the request is passed as the first argument
+     * @param callable         $first   First middleware in the chain, used to to restart a request
+     *
+     * @return Promise
+     */
+    public function handleRequest(RequestInterface $request, callable $next, callable $first)
+    {
+        $uriString = $request->getUri()->__toString();
+        if (!strpos($uriString, '://')) {
```

Comment:
Don'tt think it's possible to have something like `://example.com/foo` since it's coming from an uri object. So it would be a defectuous implementation and not our part IMO